antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -106,6 +107,17 @@ def __init__(
         self.command_factory = command_factory
         self.generator = VariantCommandGenerator(self.study_factory)
 
+    @staticmethod
+    def _get_user_name_from_id(user_id: int) -> str:
+        """
+        Utility method that retrieves a user's name based on their id.
+        Args:
+            user_id: user id (user must exist)
+        Returns: String representing the user's name
+        """
+        user_obj = db.session.query(Identity).get(user_id)
+        return t.cast(str, user_obj.name)
```

Comment:
Why do you have to cast to str ?

Reply:
Mypy cannot know the type of user_obj.name. Maybe I have to do it another way